### PR TITLE
Add permissions details section to setup

### PR DESCRIPTION
**Pull request by Charles-Gagnon**
Closes https://github.com/Azure/azure-functions-sql-extension/issues/608

### CODE REVIEW

File: docs/GeneralSetup.md [hunk 1]
```diff
@@ -42,6 +42,80 @@ ALTER TABLE ['{table_name}'] ALTER COLUMN ['{primary_key_column_name}'] int NOT
...
 
+## Create Login and User
+
+SQL bindings connect to the target database by using a Connection String configured in the app settings. This will require a login be created that the function will use to access the server.
+
+For local testing and development using a SQL (username/password) or Azure Active Directory Login is typically the easiest, but for deployed function apps it is recommended to use [Azure Active Directory Managed Authentication](https://learn.microsoft.com/azure/azure-functions/functions-identity-access-azure-sql-with-managed-identity).
+
+## Assign Permissions
+
+The login used by the function will need to have the following permissions assigned to the user it's mapped to in order for it to function. The permissions required for each type of binding is listed below.
```
Comment (dzsquared) [inline docs/GeneralSetup.md:53]: nit: using function as a verb in docs talking about the function noun, suggesting revised wording for clarity

The login used by the function will need to have the following permissions assigned to the user it's mapped to in order for it to _function_.

-->

The login used by the function will need to have the following permissions assigned to the database user it's mapped to in order for it to successfully interact with the database.

File: docs/GeneralSetup.md [hunk 2]
```diff
@@ -42,6 +42,80 @@ ALTER TABLE ['{table_name}'] ALTER COLUMN ['{primary_key_column_name}'] int NOT
...
+For stored procedure input bindings you will need `EXECUTE` permissions on the stored procedure.
+
+```sql
+USE <DatabaseName>
+GRANT EXECUTE ON <StoredProcedureName> TO <UserName>
+```
+
+### Output Binding Permissions
+
+- `SELECT`, `INSERT`, `UPDATE` and `DELETE` permissions on the table
```
Comment (lucyzhang929) [inline docs/GeneralSetup.md:79]: Is `DELETE` needed?

File: docs/GeneralSetup.md [hunk 3]
```diff
@@ -42,6 +42,80 @@ ALTER TABLE ['{table_name}'] ALTER COLUMN ['{primary_key_column_name}'] int NOT
...
+
+```sql
+USE <DatabaseName>
+GRANT CREATE SCHEMA TO <UserName>
+GRANT CREATE TABLE TO <UserName>
+```
+
+- `SELECT` and `VIEW CHANGE TRACKING` permissions on the table
+
+These are required to retrieved the data about the changes occurring in the table.
```
Comment (chlafreniere) [inline docs/GeneralSetup.md:102]:
```suggestion
These are required to retrieve the data about the changes occurring in the table.
```

File: docs/GeneralSetup.md [hunk 4]
```diff
@@ -42,6 +42,80 @@ ALTER TABLE ['{table_name}'] ALTER COLUMN ['{primary_key_column_name}'] int NOT
...
+```sql
+USE <DatabaseName>
+GRANT EXECUTE ON <StoredProcedureName> TO <UserName>
+```
+
+### Output Binding Permissions
+
+- `SELECT`, `INSERT`, `UPDATE` and `DELETE` permissions on the table
+
+These are required to retrieve metadata and update the rows in the table.
```
Comment (dzsquared) [inline docs/GeneralSetup.md:81]: we may want to add a qualifying note to the output binding:

> [NOTE]
> In some scenarios, the presence of table components such as a [SQL DML trigger](https://learn.microsoft.com/sql/relational-databases/triggers/dml-triggers) may require additional permissions for the output binding to successfully complete the operation.

**Review by dzsquared — APPROVED** [review]

File: docs/GeneralSetup.md (hunk 3 above)
Comment (MaddyDev) [reply docs/GeneralSetup.md:104]: This is still not addressed.

Comment (Charles-Gagnon) [reply docs/GeneralSetup.md]: Huh, guess I didn't click the second time. Thanks!

**Review by MaddyDev — APPROVED** [review]

**Pull request merged**